src/logic/Application.swift

```diff
@@ -158,20 +158,23 @@ class Application: NSObject {
 
     private func observeEvents() {
         guard let axObserver = axObserver else { return }
-        for notification in Application.notifications(runningApplication) {
-            retryAxCallUntilTimeout { [weak self] in
-                guard let self = self else { return }
-                try self.axUiElement!.subscribeToNotification(axObserver, notification, {
-                    DispatchQueue.main.async { [weak self] in
-                        guard let self = self else { return }
-                        // some apps have `isFinishedLaunching == true` but are actually not finished, and will return .cannotComplete
-                        // we consider them ready when the first subscription succeeds, and list their windows again at that point
-                        if !self.isReallyFinishedLaunching {
-                            self.isReallyFinishedLaunching = true
-                            self.observeNewWindows()
+        // we only need to subscribe to those apps which will have windows
+        if AXUIElement.isManageable(runningApplication) {
```

Comment:
This check is a bit late. Better not even create the `Application` instance, if we think it will not spawn any window. There is already such a method to filter apps here: https://github.com/lwouis/alt-tab-macos/blob/02590579df936d6a2124707e12d76865636fe23f/src/logic/Applications.swift#L104

You could add your new filtering logic there.

Reply:
Thank you for the suggestion. I found a compensation method to solve this problem. The initial technical attempts have been made, and I'll implement the code for this later. I'm a bit busy these two weeks.